slow_query.go

```diff
@@ -11,18 +11,38 @@ import (
 )
 
 var (
-	RegexSqlString = regexp.MustCompile("'[^']+'")
-	RegexNString   = regexp.MustCompile(`([0-9])+`)
-	RegexHasShard  = regexp.MustCompile(`(?i)from\s+(\w+|"\w+")\.`)
+	RegexSqlString  = regexp.MustCompile("('[^']+')") //Everything inside strings
+	RegexHasShard   = regexp.MustCompile(`(?i)from\s+(\w+|"\w+")\.`)
+	RegexIsDateTime = regexp.MustCompile(`(\d{4}-\d\d-\d\d \d\d:\d\d:\d\d.\d{6})`)
+	RegexGuidType   = regexp.MustCompile(`(?i)(\w{3}-\w{8}-\w{4}-\w{4}-\w{4}-\w{12})`)
+	RegexIsBool     = regexp.MustCompile(`'(t|f)'`)
 )
 
 func ScrubQuery(sql string) string {
-	sql = RegexSqlString.ReplaceAllString(sql, "'XXX'")
-	sql = RegexNString.ReplaceAllString(sql, "N")
-	// The above is very aggressive filtering
-	// need to be more specific on what gets scrubbed
+	scrubbed := RegexSqlString.ReplaceAllStringFunc(sql, scrubChecker)
+	return scrubbed
+}
+
+func isWhitelisted(value string) bool {
+	// datetime
+	isDateTime := RegexIsDateTime.MatchString(value)
+
+	// different guids
+	isGuidType := RegexGuidType.MatchString(value)
+
+	// Boolean strings of t/f
+	isBool := RegexIsBool.MatchString(value)
+
+	return isDateTime || isGuidType || isBool
+}
+
+func scrubChecker(st string) string {
```

Comment:
The queries below are helpful, but I wonder if we should add a test function focusing just on the inputs/outputs of this function. Just asserting `'t' 'f'` are ok and `'ACT-...'` matches. Show examples of what gets stripped. It might be some testing overlap, but it's a more exhaustive set of unit tests than the real-world queries bleow.

Reply:
will add additional tests for these matchers